olp-cpp-sdk-authentication/src/TokenEndpointImpl.cpp

```diff
@@ -132,7 +132,16 @@ client::OlpClient::RequestBodyType GenerateClientBody(
 TokenEndpointImpl::TokenEndpointImpl(Settings settings)
     : credentials_(std::move(settings.credentials)),
       settings_(ConvertSettings(std::move(settings))),
-      auth_client_(settings_) {}
+      auth_client_(settings_) {
+  // The underlying auth library expects a base URL and appends /oauth2/token
+  // endpoint to it. Therefore if /oauth2/token is found it is stripped from the
+  // endpoint URL provided. The underlying auth library should be updated to
+  // support an arbitrary token endpoint URL.
```

Comment:
As there is no more underlying auth library, and TokenEndpoint is the last layer, we can remove it. Instead, please add some DEBUG level logs so that we can see what the user gave us and what the parsing brought.

Reply:
done